packages/contracts/contracts/plugin/PluginRepo.sol

```diff
@@ -126,93 +144,93 @@ contract PluginRepo is IPluginRepo, Initializable, UUPSUpgradeable, ACL, Adaptiv
+    function getVersionBySemanticVersion(uint16[3] memory _semanticVersion)
         public
         view
         returns (
             uint16[3] memory semanticVersion,
-            address pluginFactoryAddress,
+            address pluginFactory,
             bytes memory contentURI
         )
     {
-        return getByVersionId(versionIdxForSemantic[semanticVersionHash(_semanticVersion)]);
+        return getVersionById(versionIndexForSemantic[semanticVersionHash(_semanticVersion)]);
     }
 
-    /// @notice get latest by version id
-    /// @return semanticVersion Semantic version for pluginRepo version
-    /// @return pluginFactoryAddress Address of plugin factory for version
-    /// @return contentURI External URI for fetching version's content
-    function getByVersionId(uint256 _versionIdx)
+    /// @notice Gets the version information associated with a version index.
+    /// @return semanticVersion The semantic version number.
+    /// @return pluginFactory The address of the plugin factory associated with the version.
+    /// @return contentURI The external URI pointing to the content of the version.
+    function getVersionById(uint256 _versionIndex)
         public
         view
         returns (
             uint16[3] memory semanticVersion,
-            address pluginFactoryAddress,
+            address pluginFactory,
             bytes memory contentURI
         )
     {
-        if (_versionIdx <= 0 || _versionIdx >= nextVersionIndex)
-            revert VersionIdxDoesNotExist({versionIdx: _versionIdx});
-        Version storage version = versions[_versionIdx];
-        return (version.semanticVersion, version.pluginFactoryAddress, version.contentURI);
+        if (_versionIndex <= 0 || _versionIndex >= nextVersionIndex)
+            revert VersionIndexDoesNotExist({versionIndex: _versionIndex});
+        Version storage version = versions[_versionIndex];
+        return (version.semanticVersion, version.pluginFactory, version.contentURI);
     }
 
-    /// @notice get version count
-    /// @return uint256 count value
+    /// @notice Gets the total number of published versions.
+    /// @return uint256 The number of published versions.
     function getVersionCount() public view returns (uint256) {
         return nextVersionIndex - 1;
     }
 
-    /// @notice check if new version is valid compared to the previous one
-    /// @param _oldVersion Old semantic version
-    /// @param _newVersion New semantic version
-    /// @return bool True if bump is valid
-    function isValidBump(uint16[3] memory _oldVersion, uint16[3] memory _newVersion)
+    /// @notice Checks if a version bump is valid.
+    /// @param _oldVersion The old semantic version number.
+    /// @param _createVersion The new semantic version number.
+    /// @return bool True if the bump is valid.
+    function isValidBump(uint16[3] memory _oldVersion, uint16[3] memory _createVersion)
```

Comment:
I think `newVersion` make much more sense than `createVersion` :warning:

Reply:
True, this renaming was unintentional and happened with the replace function.